dot/network/service.go

```diff
@@ -414,47 +417,56 @@ func (s *Service) logPeerCount() {
 	}
 }
 
-func (s *Service) publishNetworkTelemetry(done <-chan struct{}) {
+func (s *Service) publishTelemetry(cancel <-chan struct{}) {
 	ticker := time.NewTicker(s.telemetryInterval)
 	defer ticker.Stop()
 
 	for {
+		s.sentBandwidthTelemetry()
+		err := s.sendBlockIntervalTelemetry()
+		if err != nil {
+			logger.Warnf("failed to sent block interval telemetry: %s", err)
+			continue
+		}
+
 		select {
-		case <-done:
+		case <-cancel:
 			return
-
 		case <-ticker.C:
-			o := s.host.bwc.GetBandwidthTotals()
-			s.telemetry.SendMessage(telemetry.NewBandwidth(o.RateIn, o.RateOut, s.host.peerCount()))
 		}
 	}
 }
 
-func (s *Service) sentBlockIntervalTelemetry() {
-	for {
-		best, err := s.blockState.BestBlockHeader()
-		if err != nil {
-			continue
-		}
-		bestHash := best.Hash()
+func (s *Service) sentBandwidthTelemetry() {
+	o := s.host.bwc.GetBandwidthTotals()
+	s.telemetry.SendMessage(telemetry.NewBandwidth(o.RateIn, o.RateOut, s.host.peerCount()))
```

Comment:
nit rename `o` to something else. I guess `totals`?

```suggestion
	totals := s.host.bwc.GetBandwidthTotals()
	s.telemetry.SendMessage(telemetry.NewBandwidth(totals.RateIn, totals.RateOut, s.host.peerCount()))
```